# Employee Exit Surveys

## Backgroud

Staff feedback provides the companies with valuable information on the reasons why employees resign or retire. This information is used to inform attraction and retention initiatives and to improve work practices across the any company to ensure the company is considered an employer of choice.

<img src="https://fjwp.s3.amazonaws.com/blog/wp-content/uploads/2020/02/24145635/exit-interview-1024x512.png"/>

*Source*: [flexjobs](https://www.flexjobs.com/blog/post/how-to-handle-an-exit-interview/)

## Project and Data Overview

In this project, we'll work with exit surveys from employees of the [Department of Education, Training and Employment](https://en.wikipedia.org/wiki/Department_of_Education_and_Training_(Queensland)) (DETE) and the Technical and Further Education (TAFE) institude in Queensland, Australia.

The DETE Exit Survey was developed to effectively canvas the opinions and attitudes of departing employees to identify a wide range of operational, organizational and personal variables affecting the decision to leave.

We can find the DETE exit survey data [here](https://data.gov.au/dataset/ds-qld-fe96ff30-d157-4a81-851d-215f2a0fe26d/details?q=exit%20survey). However, the orignial TAFE exit survey data is no longer available.
Therefore, we'll be using the modified versions of the original datasets to make them easier to work with, which includes changing the encoding to `UTF-8` (the original ones are encoded using `cp1252`).

## Data Dictionary

The data dictionary wasn't provided with the datasets. Therefore, we'll use our general knowledge to define the columns used in them.

Below is a preview of a couple columns we'll work with from the `dete_survey.csv`:

- `ID`: An id used to identify the participant of the survey
- `SeparationType`: The reason why the person's employment ended
- `Cease Date`: The year or month the person's employment ended
- `DETE Start Date`: The year the person began employment with the DETE

Below is a preview of a couple columns we'll work with from the `tafe_survey.csv`:

- `Record ID`: An id used to identify the participant of the survey
- `Reason for ceasing employment`: The reason why the person's employment ended
- `LengthofServiceOverall. Overall Length of Service at Institute (in years)`: The length of the person's employment (in years)

## Business Problem

In this project, we'll analyze these datasets (DETE & TAFE) to find out the answers of the following questions:

- Are employees who only worked for the institues for a short period of time resigning due to some kind of dissatisfaction? What about employees who have been there longer?
- Are younger employees resigning due to some kind of dissatisfaction? What about older employees?

We'll combine the resluts for *both* surveys to answer these questions. However, although both used the same survey template, one of them customized some of the answers.


## Importing Libraries

We'll start by importing some useful libraries we need in the project.

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import matplotlib.style as style

# Display all columns in the dataframe
pd.set_option('display.max_columns', None)
# Plot style
style.use('fivethirtyeight')
# Enable the inline plotting
%matplotlib inline

## Load and Analysis Datasets

Next, we'll read in the `dete_survey.csv` and `tafe_survey.csv` datasets into pandas and explore them.

In [2]:
# Read datasets
dete_survey = pd.read_csv('dete_survey.csv')
tafe_survey = pd.read_csv('tafe_survey.csv')

Now that our datasets are loaded, we'll gather some basic information about both dataframes using `DataFrame.info()` and take a took at first few rows using `DataFrame.head()`.

## 1. DETE Survey Data

In [3]:
# Preview DETE dataset
dete_survey.info()
dete_survey.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 822 entries, 0 to 821
Data columns (total 56 columns):
 #   Column                               Non-Null Count  Dtype 
---  ------                               --------------  ----- 
 0   ID                                   822 non-null    int64 
 1   SeparationType                       822 non-null    object
 2   Cease Date                           822 non-null    object
 3   DETE Start Date                      822 non-null    object
 4   Role Start Date                      822 non-null    object
 5   Position                             817 non-null    object
 6   Classification                       455 non-null    object
 7   Region                               822 non-null    object
 8   Business Unit                        126 non-null    object
 9   Employment Status                    817 non-null    object
 10  Career move to public sector         822 non-null    bool  
 11  Career move to private sector        822 non-

ID                    SeparationType Cease Date DETE Start Date  \
0   1             Ill Health Retirement    08/2012            1984   
1   2  Voluntary Early Retirement (VER)    08/2012      Not Stated   
2   3  Voluntary Early Retirement (VER)    05/2012            2011   
3   4         Resignation-Other reasons    05/2012            2005   
4   5                    Age Retirement    05/2012            1970   

  Role Start Date                                      Position  \
0            2004                                Public Servant   
1      Not Stated                                Public Servant   
2            2011                               Schools Officer   
3            2006                                       Teacher   
4            1989  Head of Curriculum/Head of Special Education   

  Classification              Region                      Business Unit  \
0        A01-A04      Central Office  Corporate Strategy and Peformance   
1        AO5-AO7      Central Office  Corporate Strategy and Peformance   
2            NaN      Central Office               Education Queensland   
3        Primary  Central Queensland                                NaN   
4            NaN          South East                                NaN   

     Employment Status  Career move to public sector  \
0  Permanent Full-time                          True   
1  Permanent Full-time                         False   
2  Permanent Full-time                         False   
3  Permanent Full-time                         False   
4  Permanent Full-time                         False   

   Career move to private sector  Interpersonal conflicts  \
0                          False                    False   
1                          False                    False   
2                          False                    False   
3                           True                    False   
4                          False                    False   

   Job dissatisfaction  Dissatisfaction with the department  \
0                 True                                False   
1                False                                False   
2                False                                False   
3                False                                False   
4                False                                False   

   Physical work environment  Lack of recognition  Lack of job security  \
0                      False                 True                 False   
1                      False                False                 False   
2                      False                False                 False   
3                      False                False                 False   
4                      False                False                 False   

   Work location  Employment conditions  Maternity/family  Relocation  \
0          False                  False             False       False   
1          False                  False             False       False   
2          False                  False             False       False   
3          False                  False             False       False   
4          False                  False             False       False   

   Study/Travel  Ill Health  Traumatic incident  Work life balance  Workload  \
0         False       False               False              False     False   
1         False       False               False              False     False   
2         False       False               False              False     False   
3         False       False               False              False     False   
4         False       False               False               True     False   

   None of the above Professional Development Opportunities for promotion  \
0               True                        A                           A   
1              False                        A                           A   
2               True                       

Interestingly, there are some important key points in the dataset that we need to consider:

- The dataset has **822 rows** and **56 columns**.
- The column names do not conform to the recommended python naming convention rather it appears inconsistent.
- 18 out of 56 columns are stored as bool dtypes, the only column `ID` is stored as int, and rest of the columns have object dtypes.
- 32 out of 56 columns have missing values. The column `Classification` has arount 44%, whereas, `Business Unit`, `Aboriginal`, `Torres Strait`, `South Sea`, `Disability`, and `NESB` have more than 50% missing values.
- Time columns like `Cease Date`, `DETE Start Date`, and `Role Start Date` are stored as string dtypes instead of datetime.

Let's look at descriptive statistics using `describe(include='all')`. The parameter *all* with `describe()` method allows all columns to include in the output.

In [4]:
dete_survey.describe(include='all')

ID  SeparationType Cease Date DETE Start Date Role Start Date  \
count   822.000000             822        822             822             822   
unique         NaN               9         25              51              46   
top            NaN  Age Retirement       2012      Not Stated      Not Stated   
freq           NaN             285        344              73              98   
mean    411.693431             NaN        NaN             NaN             NaN   
std     237.705820             NaN        NaN             NaN             NaN   
min       1.000000             NaN        NaN             NaN             NaN   
25%     206.250000             NaN        NaN             NaN             NaN   
50%     411.500000             NaN        NaN             NaN             NaN   
75%     616.750000             NaN        NaN             NaN             NaN   
max     823.000000             NaN        NaN             NaN             NaN   

       Position Classification        Region         Business Unit  \
count       817            455           822                   126   
unique       15              8             9                    14   
top     Teacher        Primary  Metropolitan  Education Queensland   
freq        324            161           135                    54   
mean        NaN            NaN           NaN                   NaN   
std         NaN            NaN           NaN                   NaN   
min         NaN            NaN           NaN                   NaN   
25%         NaN            NaN           NaN                   NaN   
50%         NaN            NaN           NaN                   NaN   
75%         NaN            NaN           NaN                   NaN   
max         NaN            NaN           NaN                   NaN   

          Employment Status Career move to public sector  \
count                   817                          822   
unique                    5                            2   
top     Permanent Full-time                        False   
freq                    434                          800   
mean                    NaN                          NaN   
std                     NaN                          NaN   
min                     NaN                          NaN   
25%                     NaN                          NaN   
50%                     NaN                          NaN   
75%                     NaN                          NaN   
max                     NaN                          NaN   

       Career move to private sector Interpersonal conflicts  \
count                            822                     822   
unique                             2                       2   
top                            False                   False   
freq                             742                     788   
mean                             NaN                     NaN   
std                              NaN                     NaN   
min                              NaN                     NaN   
25%                              NaN                     NaN   
50%                              NaN                     NaN   
75%                              NaN                     NaN   
max                              NaN                     NaN   

       Job dissatisfaction Dissatisfaction with the department  \
count                  822                                 822   
unique                   2                                   2   
top                  False                               False   
freq                   733                                 761   
mean                   NaN                                 NaN   
std                    NaN                                 NaN   
min                    NaN                                 NaN   
25%                    NaN                                 NaN   
50%                    NaN                                 NaN   
75%                    NaN                          

Here `decribe(include='all)` method helps us further to get clarity and better understanding about our data:

- The most frequent reason why employees exit from DETE appears to be *age retirement*, as seen in the `SeparationType` column.
- Most employees are *61 or older* in the `Age` column which further solidify the reason why employees exit from DETE.
- `DETE Start Date` and `Role Start Date` columns have *Not Stated* as most frequent value and it should be referred as **NaN**.
- Columns like `Aboriginal`, `Torres Strait`, `South Sea`, `Disability`, and `NESB` have one unique value which is **Yes** and all other values are stored as **NaN**, rather they should be stored as **No**. This is the reason why these columns have such a high percentage of missing values.
- The columns from `Professional Development` to `Health & Safety` have *A* as most common values. This seems quite unusual as 'A' doesn't seem to represent anything. We'll explore these columns further.

To investigate the unusual entries like **'A'** in the columns from `Professional Development` to `Health & Safety`, we'll have to find all the unique values and count them. For this purpose, first we'll use pandas `DataFrame.apply()` on our target columns and then we'll count unique values using `lambda` function.

In [5]:
dete_survey.loc[:, 'Professional Development':'Health & Safety'].apply(lambda x: x.value_counts())

Professional Development  Opportunities for promotion  Staff morale  \
A                        413                          242           335   
D                         60                           83           112   
M                         15                           24            13   
N                        103                          230           158   
SA                       184                          100           121   
SD                        33                           56            77   

    Workplace issue  Physical environment  Worklife balance  \
A               357                   467               359   
D                77                    61               107   
M                14                    15                12   
N               160                    99               116   
SA              115                   148               162   
SD               65                    27                59   

    Stress and pressure support  Performance of supervisor  Peer support  \
A                           342                        349           401   
D                            95                         77            37   
M                            14                         12            11   
N                           168                        120            95   
SA                          124                        179           243   
SD                           67                         76            25   

    Initiative  Skills  Coach  Career Aspirations  Feedback  Further PD  \
A          396     372    345                 246       348         293   
D           34      59     65                 108        78          77   
M           13      11     22                  17        15          13   
N           95      94    141                 183       138         179   
SA         244     228    157                 130       156         149   
SD          31      47     37                  62        57          57   

    Communication  My say  Information  Kept informed  Wellness programs  \
A             399     400          436            401                253   
D              76      52           45             60                105   
M               8      10           11             10                 33   
N             129     116          120            130                225   
SA            144     177          165            162                 78   
SD             58      57           39             50                 72   

    Health & Safety  
A               386  
D                50  
M                28  
N               153  
SA              141  
SD               35

### Observation

- In our target columns, there are 6 unique values are used as likert scale: `SA`, `M`, `A`, `N`, `D`, `SD`.
- It is hard to interpret what the `M` means in these columns. Since the data source has not provided the description, we'll assume that it is likely to be abbreviated as **Missing** (i.e. the individual did not select an option). 
- The entries in these columns abbreviate as: **Strongly Agree (SA)**, **Agree (A)**, **Neutral (N)**, **Disagree (D)**, and **Strongly Disagree (SD)**.
- The most common values are `A` in every column.

## 2. TAFE Survey Data

In [8]:
# Preview TAFE dataset
tafe_survey.info()
tafe_survey.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 702 entries, 0 to 701
Data columns (total 72 columns):
 #   Column                                                                                                                                                         Non-Null Count  Dtype  
---  ------                                                                                                                                                         --------------  -----  
 0   Record ID                                                                                                                                                      702 non-null    float64
 1   Institute                                                                                                                                                      702 non-null    object 
 2   WorkArea                                                                                                                                  

Record ID                              Institute  \
0  6.341330e+17  Southern Queensland Institute of TAFE   
1  6.341337e+17            Mount Isa Institute of TAFE   
2  6.341388e+17            Mount Isa Institute of TAFE   
3  6.341399e+17            Mount Isa Institute of TAFE   
4  6.341466e+17  Southern Queensland Institute of TAFE   

                   WorkArea  CESSATION YEAR Reason for ceasing employment  \
0  Non-Delivery (corporate)          2010.0              Contract Expired   
1  Non-Delivery (corporate)          2010.0                    Retirement   
2       Delivery (teaching)          2010.0                    Retirement   
3  Non-Delivery (corporate)          2010.0                   Resignation   
4       Delivery (teaching)          2010.0                   Resignation   

  Contributing Factors. Career Move - Public Sector   \
0                                                NaN   
1                                                  -   
2                                                  -   
3                                                  -   
4                                                  -   

  Contributing Factors. Career Move - Private Sector   \
0                                                NaN    
1                                                  -    
2                                                  -    
3                                                  -    
4                       Career Move - Private Sector    

  Contributing Factors. Career Move - Self-employment  \
0                                                NaN    
1                                                  -    
2                                                  -    
3                                                  -    
4                                                  -    

  Contributing Factors. Ill Health Contributing Factors. Maternity/Family  \
0                              NaN                                    NaN   
1                                -                                      -   
2                                -                                      -   
3                                -                                      -   
4                                -                                      -   

  Contributing Factors. Dissatisfaction  \
0                                   NaN   
1                                     -   
2                                     -   
3                                     -   
4                                     -   

  Contributing Factors. Job Dissatisfaction  \
0                                       NaN   
1                                         -   
2                                         -   
3                                         -   
4                                         -   

  Contributing Factors. Interpersonal Conflict Contributing Factors. Study  \
0                                          NaN                         NaN   
1                                            -                           -   
2                                            -                           -   
3                                            -                           -   
4                                            -                           -   

  Contributing Factors. Travel Contributing Factors. Other  \
0                          NaN                         NaN   
1                       Travel                           -   
2                            -                           -   
3                       Travel                           -   
4                            -                           -   

  Contributing Factors. NONE  \
0                        NaN   
1                          -   
2                       NONE   
3                          -   
4                          -   

  Main Factor. Which of these was the main factor for leaving?  \
0                                                NaN             

We notice that the presentation of the `tafe_survey` dataset is a lot more messier than the `dete_survey` dataset in contrast, we makes it harder to analyze the data. We need to pay careful attention to extract the key points from this dataset:

- The columns do not follow Python naming convention, infact, the column names are too long and wordly.
- The dataset consists of **702 rows** and **72 columns**.
- 70 out of 72 columns are stored as string and only two columns (`Record ID` and `CESSATION YEAR`) are stored as float.
- Missing values in the **Contributing Factors** columns is 38% altogether, and the column `Main Factor. Which of these was the main factor for leaving?` has the highest about 84% of missing values in the whole dataset.
- Although these columns are presented differently, some of the columns are similar to the DETE dataset. Such as, `CESSATION YEAR`, `Reason for ceasing employment`, `Gender`, `CurrentAge`, and `EmploymentType`.

Let's dig deeper and gather some more insights with the help of `describe(include='all')`.

In [42]:
tafe_survey.describe(include='all')

Record ID                         Institute  \
count   7.020000e+02                               702   
unique           NaN                                12   
top              NaN  Brisbane North Institute of TAFE   
freq             NaN                               161   
mean    6.346026e+17                               NaN   
std     2.515071e+14                               NaN   
min     6.341330e+17                               NaN   
25%     6.343954e+17                               NaN   
50%     6.345835e+17                               NaN   
75%     6.348005e+17                               NaN   
max     6.350730e+17                               NaN   

                        WorkArea  CESSATION YEAR  \
count                        702      695.000000   
unique                         2             NaN   
top     Non-Delivery (corporate)             NaN   
freq                         432             NaN   
mean                         NaN     2011.423022   
std                          NaN        0.905977   
min                          NaN     2009.000000   
25%                          NaN     2011.000000   
50%                          NaN     2011.000000   
75%                          NaN     2012.000000   
max                          NaN     2013.000000   

       Reason for ceasing employment  \
count                            701   
unique                             6   
top                      Resignation   
freq                             340   
mean                             NaN   
std                              NaN   
min                              NaN   
25%                              NaN   
50%                              NaN   
75%                              NaN   
max                              NaN   

       Contributing Factors. Career Move - Public Sector   \
count                                                 437   
unique                                                  2   
top                                                     -   
freq                                                  375   
mean                                                  NaN   
std                                                   NaN   
min                                                   NaN   
25%                                                   NaN   
50%                                                   NaN   
75%                                                   NaN   
max                                                   NaN   

       Contributing Factors. Career Move - Private Sector   \
count                                                 437    
unique                                                  2    
top                                                     -    
freq                                                  336    
mean                                                  NaN    
std                                                   NaN    
min                                                   NaN    
25%                                                   NaN    
50%                                                   NaN    
75%                                                   NaN    
max                                                   NaN    

       Contributing Factors. Career Move - Self-employment  \
count                                                 437    
unique                                                  2    
top                                                     -    
freq                                                  420    
mean                                                  NaN    
std                                                   NaN    
min                                                   NaN    
25%                                                   NaN    
50%                                                   NaN    
75%                                                   NaN    
max                                             

There are various issues in the data that are important to highligth:

- The entries recorded in `Contributing Factors` columns have `-`. This could be indicating that no answer was provided at the time the survey was administered.
- The most frequent entry in the `Main Factor. Which of these was the main factor for leaving?` column is *Dissatisfaction with Institute* which appears *23* times only. We have to consider that this column has over *83%* of missing values as well.
- `Reason for ceasing employment` shows the reason why most employees leave the work and that is *Resignation*.
- Columns like `InstituteViews` and `WorkUnitViews` have recorded *Agree* most of the time. Similiar to *DETE* data.
- The `CurrentAge. Current Age` column has several age bins. Most of them are 56 years or older.

### Brief Description of the Oberservations

From our initial work, we can first make the following observations:

- The `dete_survey` dataframe contains `Not Stated` values that indicate values are missing, but they aren't represented as `NaN`.
- Both the `dete_survey` and `tafe_survey` dataframes contain many columns that we don't need to complete our analysis.
- Each dataframe contains many of the same columns, but the column names are different.
- There are multiple columns/answers that indicate an employee resigned because they were dissatisfied.

Let's take care of these issues next.

## Data Cleaning

To start, we'll handle the first two issues. We can use the `pd.read_csv()` function to specify values that should be represented as `NaN`. We'll use this function to fix the missing values first. Then, we'll drop columns we know we don't need for our analysis.

## 1. Specify 'Not_Stated' as 'NaN'

We read the `dete_survey.csv` file into pandas again, but this time read the `Not Stated` values in as `NaN`:

- To read `Not Stated` in as `NaN`, we'll set the `na_values` parameter to `Not Stated` in the `pd.read_csv()` funtion.

In [85]:
dete_survey = pd.read_csv('dete_survey.csv', na_values='Not Stated')
dete_survey.head()

ID                    SeparationType Cease Date  DETE Start Date  \
0   1             Ill Health Retirement    08/2012           1984.0   
1   2  Voluntary Early Retirement (VER)    08/2012              NaN   
2   3  Voluntary Early Retirement (VER)    05/2012           2011.0   
3   4         Resignation-Other reasons    05/2012           2005.0   
4   5                    Age Retirement    05/2012           1970.0   

   Role Start Date                                      Position  \
0           2004.0                                Public Servant   
1              NaN                                Public Servant   
2           2011.0                               Schools Officer   
3           2006.0                                       Teacher   
4           1989.0  Head of Curriculum/Head of Special Education   

  Classification              Region                      Business Unit  \
0        A01-A04      Central Office  Corporate Strategy and Peformance   
1        AO5-AO7      Central Office  Corporate Strategy and Peformance   
2            NaN      Central Office               Education Queensland   
3        Primary  Central Queensland                                NaN   
4            NaN          South East                                NaN   

     Employment Status  Career move to public sector  \
0  Permanent Full-time                          True   
1  Permanent Full-time                         False   
2  Permanent Full-time                         False   
3  Permanent Full-time                         False   
4  Permanent Full-time                         False   

   Career move to private sector  Interpersonal conflicts  \
0                          False                    False   
1                          False                    False   
2                          False                    False   
3                           True                    False   
4                          False                    False   

   Job dissatisfaction  Dissatisfaction with the department  \
0                 True                                False   
1                False                                False   
2                False                                False   
3                False                                False   
4                False                                False   

   Physical work environment  Lack of recognition  Lack of job security  \
0                      False                 True                 False   
1                      False                False                 False   
2                      False                False                 False   
3                      False                False                 False   
4                      False                False                 False   

   Work location  Employment conditions  Maternity/family  Relocation  \
0          False                  False             False       False   
1          False                  False             False       False   
2          False                  False             False       False   
3          False                  False             False       False   
4          False                  False             False       False   

   Study/Travel  Ill Health  Traumatic incident  Work life balance  Workload  \
0         False       False               False              False     False   
1         False       False               False              False     False   
2         False       False               False              False     False   
3         False       False               False              False     False   
4         False       False               False               True     False   

   None of the above Professional Development Opportunities for promotion  \
0               True                        A                           A   
1              False                        A                           A   
2               True           

## 2. Drop Unwanted Columns

Next, let's drop some columns from each dataframe that we won't use in our analysis to make the dataframes easier to work with:

- We'll use `DataFrame.drop()` method to drop the columns from `Professional Development` (**column 28**) to `Health & Safety` (**column 48**) in `dete_survey`. 
    - Then we'll assign the result to `dete_survey_updated`.

In [90]:
dete_unwanted_cols = dete_survey.columns[28:49]
dete_survey_updated = dete_survey.drop(dete_unwanted_cols, axis=1)
dete_survey_updated.head()

ID                    SeparationType Cease Date  DETE Start Date  \
0   1             Ill Health Retirement    08/2012           1984.0   
1   2  Voluntary Early Retirement (VER)    08/2012              NaN   
2   3  Voluntary Early Retirement (VER)    05/2012           2011.0   
3   4         Resignation-Other reasons    05/2012           2005.0   
4   5                    Age Retirement    05/2012           1970.0   

   Role Start Date                                      Position  \
0           2004.0                                Public Servant   
1              NaN                                Public Servant   
2           2011.0                               Schools Officer   
3           2006.0                                       Teacher   
4           1989.0  Head of Curriculum/Head of Special Education   

  Classification              Region                      Business Unit  \
0        A01-A04      Central Office  Corporate Strategy and Peformance   
1        AO5-AO7      Central Office  Corporate Strategy and Peformance   
2            NaN      Central Office               Education Queensland   
3        Primary  Central Queensland                                NaN   
4            NaN          South East                                NaN   

     Employment Status  Career move to public sector  \
0  Permanent Full-time                          True   
1  Permanent Full-time                         False   
2  Permanent Full-time                         False   
3  Permanent Full-time                         False   
4  Permanent Full-time                         False   

   Career move to private sector  Interpersonal conflicts  \
0                          False                    False   
1                          False                    False   
2                          False                    False   
3                           True                    False   
4                          False                    False   

   Job dissatisfaction  Dissatisfaction with the department  \
0                 True                                False   
1                False                                False   
2                False                                False   
3                False                                False   
4                False                                False   

   Physical work environment  Lack of recognition  Lack of job security  \
0                      False                 True                 False   
1                      False                False                 False   
2                      False                False                 False   
3                      False                False                 False   
4                      False                False                 False   

   Work location  Employment conditions  Maternity/family  Relocation  \
0          False                  False             False       False   
1          False                  False             False       False   
2          False                  False             False       False   
3          False                  False             False       False   
4          False                  False             False       False   

   Study/Travel  Ill Health  Traumatic incident  Work life balance  Workload  \
0         False       False               False              False     False   
1         False       False               False              False     False   
2         False       False               False              False     False   
3         False       False               False              False     False   
4         False       False               False               True     False   

   None of the above  Gender          Age Aboriginal Torres Strait South Sea  \
0               True    Male        56-60        NaN           NaN       NaN   
1              False    Male        56-60        NaN           NaN       NaN   
2               True  

Now, we'll repeat the same steps for `tafe_survey`:

- Drop the columns from `Main Factor. Which of these was the main factor for leaving?` (**column 17**) to `Workplace. Topic:Would you recommend the Institute as an employer to others?` (**column 65**).
    - Assign the result to `tafe_survey_updated`.

In [94]:
tafe_unwanted_cols = tafe_survey.columns[17:66]
tafe_survey_updated = tafe_survey.drop(tafe_unwanted_cols, axis=1)
tafe_survey_updated.head()

Record ID                              Institute  \
0  6.341330e+17  Southern Queensland Institute of TAFE   
1  6.341337e+17            Mount Isa Institute of TAFE   
2  6.341388e+17            Mount Isa Institute of TAFE   
3  6.341399e+17            Mount Isa Institute of TAFE   
4  6.341466e+17  Southern Queensland Institute of TAFE   

                   WorkArea  CESSATION YEAR Reason for ceasing employment  \
0  Non-Delivery (corporate)          2010.0              Contract Expired   
1  Non-Delivery (corporate)          2010.0                    Retirement   
2       Delivery (teaching)          2010.0                    Retirement   
3  Non-Delivery (corporate)          2010.0                   Resignation   
4       Delivery (teaching)          2010.0                   Resignation   

  Contributing Factors. Career Move - Public Sector   \
0                                                NaN   
1                                                  -   
2                                                  -   
3                                                  -   
4                                                  -   

  Contributing Factors. Career Move - Private Sector   \
0                                                NaN    
1                                                  -    
2                                                  -    
3                                                  -    
4                       Career Move - Private Sector    

  Contributing Factors. Career Move - Self-employment  \
0                                                NaN    
1                                                  -    
2                                                  -    
3                                                  -    
4                                                  -    

  Contributing Factors. Ill Health Contributing Factors. Maternity/Family  \
0                              NaN                                    NaN   
1                                -                                      -   
2                                -                                      -   
3                                -                                      -   
4                                -                                      -   

  Contributing Factors. Dissatisfaction  \
0                                   NaN   
1                                     -   
2                                     -   
3                                     -   
4                                     -   

  Contributing Factors. Job Dissatisfaction  \
0                                       NaN   
1                                         -   
2                                         -   
3                                         -   
4                                         -   

  Contributing Factors. Interpersonal Conflict Contributing Factors. Study  \
0                                          NaN                         NaN   
1                                            -                           -   
2                                            -                           -   
3                                            -                           -   
4                                            -                           -   

  Contributing Factors. Travel Contributing Factors. Other  \
0                          NaN                         NaN   
1                       Travel                           -   
2                            -                           -   
3                       Travel                           -   
4                            -                           -   

  Contributing Factors. NONE Gender. What is your Gender?  \
0                        NaN                       Female   
1                          -                          NaN   
2                       NONE                          NaN   
3                          -                          NaN   
4                  

Let's varify the changes we have made after dropping the columns in both datasets. 

In [109]:
print(f'Number of columns in DETE dataset\nbefore cleaning: {dete_survey.shape[1]}\tafter cleaning: {dete_survey_updated.shape[1]}')
print('-'*40)
print(f'Number of columns TAFE dataset\nbefore cleaning: {tafe_survey.shape[1]}\tafter cleaning: {tafe_survey_updated.shape[1]}')

Number of columns in DETE dataset
before cleaning: 56	after cleaning: 35
----------------------------------------
Number of columns TAFE dataset
before cleaning: 72	after cleaning: 23
